source/helpers/on-replaced-element.ts

```diff
@@ -25,7 +25,11 @@ export default async function onReplacedElement(
 
 	while (trackedElement) {
 		// eslint-disable-next-line no-await-in-loop
-		await onElementRemoval(trackedElement);
+		await onElementRemoval(trackedElement, signal);
+		if (signal?.aborted) {
+			return;
+		}
```

Comment:
This part feels really awkward, because what if the signal is aborted before the `while` loop? Or just before the callback is called?

But short of putting `if` blocks everywhere, I don't really know how to improve it :thinking:

Reply:
I think it's ok, `onElementRemoval` is not a poller